node/module.js

```diff
@@ -0,0 +1,50 @@
+/*
+ * # Copyright (c) 2016-2017 The Khronos Group Inc.
+ * # Copyright (c) 2016 Alexey Knyazev
+ * #
+ * # Licensed under the Apache License, Version 2.0 (the "License");
+ * # you may not use this file except in compliance with the License.
+ * # You may obtain a copy of the License at
+ * #
+ * #     http://www.apache.org/licenses/LICENSE-2.0
+ * #
+ * # Unless required by applicable law or agreed to in writing, software
+ * # distributed under the License is distributed on an "AS IS" BASIS,
+ * # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * # See the License for the specific language governing permissions and
+ * # limitations under the License.
+ */
+
+import validator from './gltf_validator.dart.js';
+
+/**
+ * Validates an asset from bytes.
+ * @param {Uint8Array} data - Byte array containing glTF or GLB data.
+ * @param {ValidationOptions} options - Object with validation options.
+ * @returns {Promise} Promise with validation result in object form.
+ */
+export const validateBytes = (data, options) => validator.validateBytes(data, options);
+
+/**
+ * Validates an asset from JSON string.
+ * @param {string} json - String containing glTF JSON.
+ * @param {ValidationOptions} options - Object with validation options.
+ * @returns {Promise} Promise with validation result in object form.
+ */
+export const validateString = (json, options) => validator.validateString(json, options);
+
+/**
+ @typedef {Object} ValidationOptions
+ @property {string} uri - Absolute or relative asset URI that will be copied to validation report.
+ @property {ExternalResourceFunction} externalResourceFunction - Function for loading external resources. If omitted, external resources are not validated.
+ @property {boolean} validateAccessorData - Set to `false` to skip reading of accessor data.
+ @property {number} maxIssues - Max number of reported issues. Use `0` for unlimited output.
+ @property {string[]} ignoredIssues - Array of ignored issue codes.
+ @property {Object} severityOverrides - Object with overridden severities for issue codes.
+ */
+
+/**
+ * @callback ExternalResourceFunction
+ * @param {string} uri - Relative URI of the external resource.
+ * @returns {Promise} - Promise with Uint8Array data.
+ */
```

Comment:
if we're only going to generate documentation from `index.js`, I don't think I'd keep any more JSDoc here than necessary. Particularly the ValidationOptions and ExternalResourceFunction definitions may be inconvenient to keep in sync.

Reply:
I think we need those definitions because exported functions' JSDoc mentions them. May be there's a more uniform way of documenting public JS API? Anyway, it's not like doc changes often, so probably we can leave that as is.

Reply:
I think of the [API docs on npm](https://www.npmjs.com/package/gltf-validator#api) as the documentation people would read, and the JSDoc in `index.js` as an incidental requirement to generate that, so further JSDoc isn't strictly adding anything from my perspective, but if you're happy to keep it in sync I'm not opposed either.